taipy/core/common/reason.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2021-2024 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from typing import Dict, Set
+
+
+class Reason:
```

Comment:
Do we intend to use this class anywhere else?

The `Reason` naming seems to vague to me.
```suggestion
class BlockingReason:
```
or
```suggestion
class EditBlockingReason:
```

Reply:
Due to the purpose for generic use of this class, I think we should keep it as `Reason`